File: FluentApiGateway/src/main/kotlin/com/kandrac/matej/fluentapigateway/dto/VideoUploadResponse.kt
package com.kandrac.matej.fluentapigateway.dto

data class VideoUploadResponse(
    val success: Boolean,
    val message: String,
    val videoId: Long? = null,
    val filename: String? = null,
    val fileSize: Long? = null,
    val uploadedAt: String? = null
)

File: fluent/lib/api/models/video_upload_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'video_upload_response.g.dart';

@JsonSerializable()
class VideoUploadResponse {
  final bool success;
  final String message;
  final String? id;
  final String? filename;
  final int? fileSize;
  final String? uploadedAt;

  VideoUploadResponse({
    required this.success,
    required this.message,
    this.id,
    this.filename,
    this.fileSize,
    this.uploadedAt,
  });

  factory VideoUploadResponse.fromJson(Map<String, dynamic> json) =>
      _$VideoUploadResponseFromJson(json);

  Map<String, dynamic> toJson() => _$VideoUploadResponseToJson(this);
}